textx/model.py

```diff
@@ -195,9 +217,14 @@ def get_model_from_file(self, file_name, encoding, debug):
             with codecs.open(file_name, 'r', encoding) as f:
                 model_str = f.read()
 
-            model = self.get_model_from_str(model_str, file_name=file_name,
-                                            debug=debug)
+            model = self.get_model_from_str(
+                model_str, file_name=file_name, debug=debug,
+                pre_ref_resolution_callback=pre_ref_resolution_callback,
+                is_main_model=is_main_model)
 
+            # reset the file: see "# Register filename of the model for later
+            # use (e.g. imports/scoping)."
+            # w/o this second assignment some tests fail (TBC)
```

Comment:
What does this comment mean?

Reply:
I removed the comment. And I removed the code below, too (model._tx_filename=...). Is also adapted the "model from str" accordingly. The filename is set just before reference resolution (after converting the parse tree). This is mandatory, since some scope providers need to know the filename of the model.